controller/contour/deployment.go

```diff
@@ -0,0 +1,423 @@
+
+	certsVol := corev1.Volume{
+		Name: certsVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: certsSecretName,
+			},
+		},
+	}
+	certsVolMt := corev1.VolumeMount{
+		Name:      certsVolumeName,
+		MountPath: certsVolumeMountDir,
+		ReadOnly:  true,
+	}
+	cfgMapPerms := int32(0644)
+	cfgVol := corev1.Volume{
+		Name: cfgVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: deployCfgMapName,
+				},
+				Items:       []corev1.KeyToPath{{Key: cfgFileName, Path: cfgFileName}},
+				DefaultMode: &cfgMapPerms,
+			},
+		},
+	}
+	cfgVolMt := corev1.VolumeMount{
+		Name:      cfgVolumeName,
+		MountPath: cfgVolumeMountDir,
+		ReadOnly:  true,
+	}
+	deploy.Spec.Template.Spec.Volumes = []corev1.Volume{certsVol, cfgVol}
+	deploy.Spec.Template.Spec.Containers[0].VolumeMounts = []corev1.VolumeMount{certsVolMt, cfgVolMt}
+
+	user := int64(65534)
+	group := int64(65534)
+	nonRoot := true
+	secCtx := corev1.PodSecurityContext{
+		RunAsUser:    &user,
+		RunAsGroup:   &group,
+		RunAsNonRoot: &nonRoot,
+	}
+	deploy.Spec.Template.Spec.SecurityContext = &secCtx
+
+	deploy.Spec.Template.Spec.Affinity = &corev1.Affinity{
+		PodAffinity: &corev1.PodAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+				{
+					Weight: int32(100),
+					PodAffinityTerm: corev1.PodAffinityTerm{
+						TopologyKey: "kubernetes.io/hostname",
+						LabelSelector: &metav1.LabelSelector{
+							MatchLabels: labels,
+						},
+					},
+				},
+			},
+		},
+	}
+	// TODO [danehans]: Readdress anti-affinity when https://github.com/projectcontour/contour/issues/2997
+	// is resolved.
+
+	return deploy, nil
+}
+
+// currentDeployment returns the current Deployment resource for the provided contour.
+func (r *Reconciler) currentDeployment(ctx context.Context, contour *operatorv1alpha1.Contour) (bool, *appsv1.Deployment, error) {
+	deploy := &appsv1.Deployment{}
+	key := types.NamespacedName{
+		Namespace: contour.Spec.Namespace.Name,
+		Name:      contour.Name,
+	}
+
+	if err := r.Client.Get(ctx, key, deploy); err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil, nil
+		}
+		return false, nil, err
+	}
+
+	return true, deploy, nil
+}
+
+// createDeployment creates a Deployment resource for the provided deploy.
+func (r *Reconciler) createDeployment(ctx context.Context, deploy *appsv1.Deployment) error {
+	if err := r.Client.Create(ctx, deploy); err != nil {
+		return fmt.Errorf("failed to create deployment %s/%s: %v", deploy.Namespace, deploy.Name, err)
+	}
+	r.Log.Info("created deployment", "namespace", deploy.Namespace, "name", deploy.Name)
+
+	return nil
+}
+
+// updateDeployment updates a Deployment if current does not match desired.
+func (r *Reconciler) updateDeployment(ctx context.Context, current, desired *appsv1.Deployment) (bool, error) {
```

Comment:
Why does this need to return the bool as well? Seems like we don't need to care whether there was an update performed, as long as we know the desired state was applied.